### PR TITLE
updates to the tutorials

### PR DESCRIPTION
**Pull request by djarecka**
I've only started the changes - for now I only worked on:
- `1_intro_pydra`
- `2_intro_functiontask`

I've tried to think about the Diataxis suggestions, mostly about:
- "a tutorial must help a beginner achieve basic competence with a product"
- "tutorials are learning-oriented, and how-to guides are task-oriented"

I've removed several sections thinking that they should be moved to:
- Explanations:
    - the text from `1_intro_pydra`
- How-To Guides:
    - second way of annotation the output
    - entire Audit section
    - entire Type Checking section

In addition I added several TODO-LINK to remember about the links from future Explanations (or Installation) sections. I also change "we" to "you" in most of the places.

Please let me know what you think! You can suggest changes in this PR, or if you prefer you can merge it and work on improvement starting from new master (as long as we don't have to resolve conflicts in the notebooks everything works :) )

### CODE REVIEW

File: notebooks/2_intro_functiontask.md [hunk 1]
```diff
@@ -37,159 +38,62 @@ def add_var(a, b):
     return a + b
 ```
 
-Once we decorate the function, we can create a pydra `Task` and specify the input:
+Now that we decorated the function, we can create a pydra `Task` and specify the input, for this example values for `a` and `b` are needed.
```
Comment (yibeichan) [inline notebooks/2_intro_functiontask.md:41]: how about rewording it as

After decorating the function, we can create a pydra `Task` and specify the inputs. In this example, values for `a` and `b` are needed.

File: notebooks/2_intro_functiontask.md [hunk 2]
```diff
@@ -37,159 +38,62 @@ def add_var(a, b):
...
-# This will raise a ValueError
-task1f = sum_of_pairs(pairs=[(1, 2), (3, "4")])  
-task1f()
-```
-
 ## Customizing output names
-Note, that "out" is the default name for the task output, but we can always customize it. There are two ways of doing it: using *python* function annotation and using another *pydra* decorator:
-
-Let's start from the function annotation:
+Note, that "out" is the default name for the task output, but you can always customize it by using *python* function annotation.
```
Comment (yibeichan) [inline notebooks/2_intro_functiontask.md:96]: Note, that "out" is -> Note, that `out` from `result.output.out`

File: notebooks/2_intro_functiontask.md [hunk 3]
```diff
@@ -361,33 +252,36 @@ task4c(rerun=True)
...
 
-and when we check the `output_dir`, we can see that it's different than last time:
+and when you check the `output_dir`, you can see that it's different than last time:
 
 ```{code-cell} ipython3
 task4b.output_dir
 ```
 
-This is because, the checksum changes when we change either input or function.
+This is because, the checksum changes when you change either input or function.
```
Comment (yibeichan) [inline notebooks/2_intro_functiontask.md:269]: do we need to add a link here to explain what checksum is and what it does in general?

**Review by yibeichan — COMMENTED** [review]
the content looks good to me. a few notes:
- do we want to set up a rough length limit for each notebook in the tutorial/how-to-guide/etc.
- I'm okay with using "we" when explaining things to the user; it feels like "we" are from the same group. we can use "you" when we want the user to try some exercises.

File: notebooks/2_intro_functiontask.md (hunk 3 above)
Comment (djarecka) [reply notebooks/2_intro_functiontask.md:269]: yes, i marked it earlier
